UI/volume-control.cpp

```diff
@@ -1217,7 +1222,12 @@ void VolumeMeterTimer::RemoveVolControl(VolumeMeter *meter)
 
 void VolumeMeterTimer::timerEvent(QTimerEvent *)
 {
-	// Tell paintEvent to paint only the bars, leaving the scale alone.
-	for (VolumeMeter *meter : volumeMeters)
-		meter->update(meter->getBarRect());
+	for (VolumeMeter *meter : volumeMeters) {
+		if (meter->needLayoutChange())
+			// Tell paintEvent to update layout and paint everything
+			meter->update();
+		else
+			// Tell paintEvent to paint only the bars, leaving the scale alone.
+			meter->update(meter->getBarRect());
```

Comment:
I spent 20 years working under a style guide that required curlies on ALL ifs and loops, and I confess that I prefer that style, but the guide at https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst says no braces if there is only one STATEMENT, vs one line.

So does OBS culture and custom suggest curlies here, remove the comments, or single comment above the if?

Reply:
As I mentioned in [another review comment](https://github.com/obsproject/obs-studio/pull/5233#discussion_r702337941):
> This technically doesn't need curly braces because each branch contains only one statement, but I'm not going to draw a hard line on it.

We don't hard-enforce that particular guideline in this repo.  IIRC, Jim tends to prefer always using braces these days, even though the style guide we use as a baseline says not to in certain cases.

I don't recall ever encountering an `if-else` block with branches containing single statements _along with_ comments.  Such an instance, in my opinion, warrants braces.

Otherwise, simply commenting succinctly before the entire `if-else` block would avoid the entire subject.
```
// If a layout change is needed, tell paintEvent to update layout and paint everything.
// Otherwise, tell paintEvent to paint only the bars, leaving the scale alone.
if (meter->needLayoutChange())
	meter->update();
else
	meter->update(meter->getBarRect());
```

Reply:
In short, if you use curly braces everywhere in newly submitted code, I'm not going to _insist_ that you remove them and I'm not going to hold the PR back on such a point.  I simply pointed out that they were not necessary and that you could remove them.